Microsoft.Toolkit.Uwp.UI.Controls/CameraPreview/CameraPreview.cs

```diff
@@ -0,0 +1,182 @@
+            if (result != CameraHelperResult.Success)
+            {
+                InvokePreviewFailed(result.ToString());
+            }
+
+            SetUIControls(result);
+        }
+
+        private async void FrameSourceGroupButton_ClickAsync(object sender, RoutedEventArgs e)
+        {
+            _selectedSourceIndex = _selectedSourceIndex < (_frameSourceGroups.Count - 1) ? _selectedSourceIndex + 1 : 0;
+            var group = _frameSourceGroups[_selectedSourceIndex];
+            _frameSourceGroupButton.IsEnabled = false;
+            _cameraHelper.FrameSourceGroup = group;
+            await InitializeAsync();
+        }
+
+        private void InvokePreviewFailed(string error)
+        {
+            EventHandler<PreviewFailedEventArgs> handler = PreviewFailed;
+            handler?.Invoke(this, new PreviewFailedEventArgs { Error = error });
+        }
+
+        private void SetMediaPlayerSource()
+        {
+            try
+            {
+                var frameSource = _cameraHelper?.FrameSource;
+                if (frameSource != null)
+                {
+                    if (_mediaPlayer == null)
+                    {
+                        _mediaPlayer = new MediaPlayer
+                        {
+                            AutoPlay = true,
+                            RealTimePlayback = true
+                        };
+                    }
+
+                    _mediaPlayer.Source = MediaSource.CreateFromMediaFrameSource(frameSource);
+                    _mediaPlayerElementControl.SetMediaPlayer(_mediaPlayer);
+                }
+            }
+            catch (Exception ex)
+            {
+                InvokePreviewFailed(ex.Message);
+            }
+        }
+
+        private void SetUIControls(CameraHelperResult result)
+        {
+            var success = result == CameraHelperResult.Success;
+            if (success)
+            {
+                SetMediaPlayerSource();
+            }
+            else
+            {
+                _mediaPlayerElementControl.SetMediaPlayer(null);
+            }
+
+            _frameSourceGroupButton.IsEnabled = IsFrameSourceGroupButtonAvailable;
+            SetFrameSourceGroupButtonVisibility();
+        }
+
+        private void SetFrameSourceGroupButtonVisibility()
+        {
+            _frameSourceGroupButton.Visibility = IsFrameSourceGroupButtonAvailable && IsFrameSourceGroupButtonVisible
+                                                                ? Visibility.Visible
+                                                                : Visibility.Collapsed;
+        }
+
+        /// <summary>
+        /// Dispose resources.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task CleanupAsync()
+        {
+            if (_cameraHelper != null)
+            {
+                await _cameraHelper.CleanupAsync();
```

Comment:
should the control dispose the camerahelper. What if I want to use the camerahelper across multiple pages?